packages/pdf-viewer/VerticalPages.tsx

```diff
@@ -1,42 +1,82 @@
-import React, { useLayoutEffect, useRef, useState } from 'react';
+import { useLayoutEffect, useEffect, useRef, useState, MutableRefObject } from 'react';
+import * as React from 'react';
 import { PdfData, ScaledSize } from './pdfSource';
 import Page from './Page';
+import styled from 'styled-components';
 
-require('./pages.css');
 
+const PagesHolder = styled.div<{ pageGap: number }>`
+	display: flex;
+	justify-content: center;
+	align-items: center;
+	flex-flow: column;
+	width: 100%;
+	min-height: 100%;
+	row-gap: ${(props)=> props.pageGap || 2}px;
+`;
 
 export interface VerticalPagesProps {
 	pdf: PdfData;
 	isDarkTheme: boolean;
-	anchorPage: number;
-	container: React.MutableRefObject<HTMLElement>;
+	anchorPage?: number;
+	rememberScroll?: boolean;
+	pdfId?: string;
+	container: MutableRefObject<HTMLElement>;
+	pageGap?: number;
+	showPageNumbers?: boolean;
 }
 
 
 export default function VerticalPages(props: VerticalPagesProps) {
 	const [scaledSize, setScaledSize] = useState<ScaledSize>(null);
+	const currentScaleSize = useRef(scaledSize);
 	const innerContainerEl = useRef<HTMLDivElement>(null);
+
+	const updateSize = async (cancelled?: boolean) => {
+		if (props.pdf) {
+			const containerWidth = innerContainerEl.current.clientWidth - 10;
+			const scaledSize_ = await props.pdf.getScaledSize(null, containerWidth);
+			if (cancelled) return;
+			innerContainerEl.current.style.height = `${(scaledSize_.height + (props.pageGap || 2)) * props.pdf.pageCount}px`;
+			const oldScaleSize = currentScaleSize.current;
+			if (oldScaleSize && props.container.current) {
+				const oldScrollTop = props.container.current.scrollTop;
+				props.container.current.scrollTop = oldScrollTop * (scaledSize_.scale / oldScaleSize.scale);
+			}
+			currentScaleSize.current = scaledSize_;
+			setScaledSize(scaledSize_);
+		}
+	};
+
 	useLayoutEffect(() => {
 		let resizeTimer: number = null;
 		let cancelled = false;
-		const updateSize = async () => {
-			if (props.pdf) {
-				const innerWidth = innerContainerEl.current.clientWidth;
-				const scaledSize = await props.pdf.getScaledSize(null, innerWidth - 10);
-				if (cancelled) return;
-				setScaledSize(scaledSize);
-			}
```

Comment:
You're introducing implied dependencies with this refactoring. For example you removed "props.pdf" as a dependency - why? It's still a dependency except it's now lost in the above refactored code.

If you refactor, create a custom hook that clearly lists dependencies. Please read about React Hooks and understand how they work as your lib is not going to work properly if you don't handle your dependencies well.

Reply:
Actually I skipped those deps because I don't expect them to change over the course of time (in a way they should not cause re-renders), being new to functional components I wasn't aware I should always add them too. Sorry for such ignorance.

Reply:
Yes always add them. If they don't change - great, there won't a re-render. But if later the code is updated, that dependency might change over time, and then there would be a bug here.